doc: release-notes-latest: Summarize networking changes from Zephyr (#4243)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-latest.rst

The following list summarizes the most important changes inherited from upstream Zephyr:

* Networking:

  * General:

    * Added UDP commands to the network shell.

>     * Added network interface status verification before sending packet.
>     * Added missing ``getaddrinfo()`` error codes translation.

Reviewer: ```suggestion
    * Added verification of the network interface status before sending a packet.
    * Added missing translations for ``getaddrinfo()`` error codes.
```

    * Added a separate work queue for TCP2.

>     * Added multiple bugfixes for IEEE802154 L2.

Reviewer: ```suggestion
    * Added multiple bug fixes for IEEE 802.15.4 L2.
```

    * Fixed memory management issues in TCP2 when running out of memory.
    * Added connection establishment timer for TCP2.

  * LwM2M:

>     * Fixed a bug where large LwM2M endpoint names were not encoded properly in the Register message.

Reviewer: ```suggestion
    * Fixed a bug where large LwM2M endpoint names were not encoded properly in the registration message.
```

    * Added API functions to update minimum/maximum observe period of a resource.

  * OpenThread:

>     * Updated the OpenThread version to commit ``8f7024c3e9beb47a48cfc1e3185f5fce82fffba9``.

Reviewer: Need an empty line before the sublist.

    * Added external heap implementation in OpenThread platform.
    * Removed an obsolete ``CONFIG_OPENTHREAD_NCP_BUFFER_SIZE`` option.

>     * Added new OpenThread options:

Reviewer: ```suggestion
    * Added the following OpenThread options:
```

      * :option:`CONFIG_OPENTHREAD_COAP_BLOCK`
      * :option:`CONFIG_OPENTHREAD_MASTERKEY`
      * :option:`CONFIG_OPENTHREAD_SRP_CLIENT`
      * :option:`CONFIG_OPENTHREAD_SRP_SERVER`

  * MQTT:

>     * Fixed UTF-8 strings logging.

Reviewer: ```suggestion
    * Fixed logging of UTF-8 strings.
```

  * Sockets:

>     * Fixed TLS sockets access from userspace.

Reviewer: ```suggestion
    * Fixed TLS sockets access from user space.
```

  * CoAP:

    * Added a symbol for the default COAP version.
    * Fixed a discovery response formatting.

>     * Updated a few API functions, to accept a const pointer when appropriate.

Reviewer: ```suggestion
    * Updated a few API functions to accept a const pointer when appropriate.
```
